security/Samples/SecurityProviderTpmSimulator/SecurityProviderTpmSimulator.cs

```diff
@@ -65,17 +67,22 @@ public override byte[] Sign(byte[] data)
         }
 
 
-        [SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "Sample code resiliency")]
         public static void StopSimulatorProcess()
         {
-            foreach (var process in Process.GetProcessesByName(Path.GetFileNameWithoutExtension(s_simulatorExeName)))
+            if (s_simulatorProcessId != 0)
             {
                 try
                 {
-                    process?.Kill();
+                    var process = Process.GetProcessById(s_simulatorProcessId);

```

Comment:
nit: it's clear by the method name that we are getting a Process by Id. But is it best practice to use Process instead of var?